### PR TITLE
Added label logic

### PR DESCRIPTION
**Pull request by VlasenkoYevhenii**

### CODE REVIEW

File: .env
Comment (makaroshyna) [inline .env:1]: Delete this.

File: docker-compose.yml
Comment (makaroshyna) [inline docker-compose.yml:1]: Delete this.

File: pom.xml [hunk 1]
```diff
@@ -82,6 +82,11 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-docker-compose</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+			<version>2.3.0</version>
```
Comment (makaroshyna) [inline pom.xml:88]: Add the version to properties higher.

File: src/main/java/app/controller/LabelController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/labels")
+@Tag(name = "Label", description = "Labels management")
```
Comment (makaroshyna) [inline src/main/java/app/controller/LabelController.java:25]:
```suggestion
@Tag(name = "Label management", description = "Endpoints for managing labels")
```

File: src/main/java/app/controller/LabelController.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/labels")
```
Comment (makaroshyna) [inline src/main/java/app/controller/LabelController.java:24]: Maybe without api will be better? We didn't talk about it.
```suggestion
@RequestMapping("/labels")
```

File: src/main/java/app/controller/LabelController.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+public class LabelController {
+    private final LabelService labelService;
+
+    @PostMapping
+    @Operation(summary = "Label creation", description = "Create and save a new label")
+    public LabelResponseDto createLabel(@Valid @RequestBody LabelRequestDto dto) {
+        return labelService.create(dto);
+    }
+
+    @GetMapping("/by_project/{projectId}")
```
Comment (makaroshyna) [inline src/main/java/app/controller/LabelController.java:35]: This looks strange. Maybe put this into project controller? Or remove altogether?

File: src/main/java/app/controller/LabelController.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/labels")
+@Tag(name = "Label", description = "Labels management")
+public class LabelController {
+    private final LabelService labelService;
+
+    @PostMapping
```
Comment (makaroshyna) [inline src/main/java/app/controller/LabelController.java:29]: You can add response status here.
```suggestion
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
```

File: src/main/java/app/service/LabelServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            existingLabel.setColor(dto.getColor());
+
+            Label updatedLabel = repository.save(existingLabel);
+            return mapper.toDto(updatedLabel);
+        }
+        throw new EntityNotFoundException("Failed to find entity by id=" + id);
+    }
+
+    @Override
+    public void deleteLabel(Long id) {
```
Comment (makaroshyna) [inline src/main/java/app/service/LabelServiceImpl.java:52]: Check if there is a label with this ID and then delete it.

**Review by makaroshyna — CHANGES_REQUESTED** [review]
Good! Left some comments.

File: src/main/java/app/controller/LabelController.java (hunk 4 above)
Comment (VlasenkoYevhenii) [reply src/main/java/app/controller/LabelController.java:35]: I suggest to test this when all related services will be implemented and if it doesn't work or gets lazy -delete it

File: src/main/java/app/service/LabelService.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+public interface LabelService {
+    LabelResponseDto create(LabelRequestDto dto);
+
+    Set<LabelResponseDto> getAllLabelsForProject(Long projectId);
+
+    Set<LabelResponseDto> getAllByTaskId(Long taskId);
+
+    LabelResponseDto update(Long id, LabelRequestDto dto);
+
+    void deleteLabel(Long id);
```
Comment (ValentynSharshon) [inline src/main/java/app/service/LabelService.java:16]: It may be better to use one style of method naming. For example: create or createLabel.

**Review by ValentynSharshon — CHANGES_REQUESTED** [review]

**Review by ValentynSharshon — APPROVED** [review]

**Review by makaroshyna — APPROVED** [review]

**Pull request merged**